### PR TITLE
TINY-10385: fix schema.isWrapper function

### PR DESCRIPTION
**Pull request by lorenzo-pomili**
Related Ticket: TINY-10385

Description of Changes:
backport for: https://github.com/tinymce/tinymce/pull/9195

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] ~Docs ticket created (if applicable)~

GitHub issues (if applicable):

### CODE REVIEW

**Review by HAFRMO — APPROVED** [review]

File: modules/tinymce/CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 - Translation syntax for announcement text in the table grid was incorrectly formatted. #TINY-10141
+- The functions `schema.isWrapper` and `schema.isInline` didn't exclude element names that started with `#` those should not be considered elements. #TINY-10385
```
Comment (mkzhx) [inline modules/tinymce/CHANGELOG.md:11]:
```suggestion
- The functions `schema.isWrapper` and `schema.isInline` did not exclude node names that started with `#` which should not be considered as elements. #TINY-10385
```

**Review by mkzhx — APPROVED** [review]

File: modules/tinymce/src/core/test/ts/webdriver/paste/CutTest.ts [hunk 2]
```diff
@@ -20,4 +20,23 @@ describe('webdriver.tinymce.core.paste.CutTest', () => {
     await RealMouse.pClickOn('div[title="Cut"]');
     await Waiter.pTryUntil('Cut is async now, so need to wait for content', () => TinyAssertions.assertContent(editor, '<p>ac</p>'));
   });
+
+  context('TINY-10385: cutting a text at the first level with `valid_elements` `*[*]` should not make the editor crush', () => {
```
Comment (ltrouton) [inline modules/tinymce/src/core/test/ts/webdriver/paste/CutTest.ts:24]:
```suggestion
  context('TINY-10385: cutting text at the first level with `valid_elements` `*[*]` should not make the editor crash', () => {
```

File: modules/tinymce/src/core/main/ts/api/html/Schema.ts [hunk 3]
```diff
@@ -571,7 +571,7 @@ const Schema = (settings: SchemaSettings = {}): Schema => {
 
   const isBlock = (name: string): boolean => Obj.has(getBlockElements(), name);
 
-  const isInline = (name: string): boolean => isValid(name) && !isBlock(name);
+  const isInline = (name: string): boolean => !Strings.startsWith(name, '#') && isValid(name) && !isBlock(name);
```
Comment (ltrouton) [inline modules/tinymce/src/core/main/ts/api/html/Schema.ts:574]: I feel like we should add a comment above this line explaining why we are checking for the `#`. Could easily see myself forgetting why the check is being done.
```suggestion
	// Check if name starts with # to detect non-element node names like #text and #comment
  const isInline = (name: string): boolean => !Strings.startsWith(name, '#') && isValid(name) && !isBlock(name);
```

**Review by ltrouton — APPROVED** [review]

File: modules/tinymce/CHANGELOG.md (hunk 1 above)
Comment (lorenzo-pomili) [reply modules/tinymce/CHANGELOG.md]: changed

File: modules/tinymce/src/core/test/ts/webdriver/paste/CutTest.ts (hunk 2 above)
Comment (lorenzo-pomili) [reply modules/tinymce/src/core/test/ts/webdriver/paste/CutTest.ts]: changed

File: modules/tinymce/src/core/main/ts/api/html/Schema.ts (hunk 3 above)
Comment (lorenzo-pomili) [reply modules/tinymce/src/core/main/ts/api/html/Schema.ts:575]: added

**Pull request merged**